Resources/Public/Javascript/PageView/SearchInDocument.js

```diff
@@ -43,10 +43,226 @@ function resetStart() {
+
+    var queryParam;
+    for(var i = 0; i < params.length; i++) {
+        queryParam = params[i].split('=');
+        if(queryParams.indexOf(queryParam[0]) === -1) {
+            queryParams.push(queryParam[0]);
+            queryParams[queryParam[0]] = queryParam[1];
+        }
+    }
+    return queryParams;
+}
+
+/**
+ * Get needed URL query parameters.
+ * It returns array of params as objects 'param' => 'value'. It contains exactly 3 params which are taken out of search result.
+ * 
+ * @param {array} element
+ * 
+ * @returns {array} array with params in form 'param' => 'value' 
+ */
+function getNeededQueryParams(element) {
+    var id = $("input[id='tx-dlf-search-in-document-id']").attr('name');
+    var highlightWord = $("input[id='tx-dlf-search-in-document-highlight-word']").attr('name');
+    var page = $("input[id='tx-dlf-search-in-document-page']").attr('name');
+
+    var queryParams = [];
+        
+    if(id && getBaseUrl(element['uid']).split('?')[0].indexOf(element['uid']) === -1) {
+        queryParams.push(id);
+        queryParams[id] = element['uid'];
+    }
+
+    if(highlightWord) {
+        queryParams.push(highlightWord);
+        queryParams[highlightWord] = encodeURIComponent($("input[id='tx-dlf-search-in-document-query']").val());
+    }
+
+    if(page) {
+        queryParams.push(page);
+        queryParams[page] = element['page'];
+    }
+
+    return queryParams;
+}
+
+/**
+ * Get snippet link.
+ * 
+ * @param {array} element
+ * 
+ * @returns {string}
+ */
+function getLink(element) {
+    var baseUrl = getBaseUrl(element['uid']);
+
+    var queryParams = getNeededQueryParams(element);
+
+    if (baseUrl.indexOf('?') > 0) {
+        queryParams = getAllQueryParams(baseUrl, queryParams);
+        baseUrl = baseUrl.split('?')[0];
+    }
+
+    var link = baseUrl + '?';
+
+    // add query params to result link
+    for(var i = 0; i < queryParams.length; i++) {
+        link += queryParams[i] + '=' + queryParams[queryParams[i]] + '&';
+    }
+    link = link.slice(0, -1);
+    return link;
+}
+
+function getNavigationButtons(start, numFound) {
+    var buttons = "";
+
+    if (start > 0) {
+        buttons += '<input type="button" id="tx-dlf-search-in-document-button-previous" class="button-previous" onclick="previousResultPage();" value="' + $('#tx-dlf-search-in-document-label-previous').text() + '" />';
+    }
+
+    if (numFound > (start + 20)) {
+        buttons += '<input type="button" id="tx-dlf-search-in-document-button-next" class="button-next" onclick="nextResultPage();" value="' + $('#tx-dlf-search-in-document-label-next').text() + '" />';
+    }
+    return buttons;
+}
+
+function getCurrentPage() {
+    var page = 1;
+    var queryParams = getCurrentQueryParams(getBaseUrl(" "));
+
+    for(var i = 0; i < queryParams.length; i++) {
+        var queryParam = queryParams[i].split('=');
+
+        if(queryParam[0] === $("input[id='tx-dlf-search-in-document-page']").attr('name')) {
```

Comment:
This comparison doesn't respect the possible encoding of the parameters. This is wrong on other places, too.

If you click next/last page, the link parameter get's encoded:
<code>
/workview?tx_dlf%5Bdouble%5D=0&tx_dlf%5Bhighlight_word%5D=pieschen&tx_dlf%5Bid%5D=1&tx_dlf%5Bpage%5D=3&cHash=04b8a699974161fd87b83c9b36d65483
</code>

But `'tx_dlf%5Bpage%5D' === 'tx_dlf[page]'` is not true so the highlighting fails.

Reply:
Highlighting seems to work now. Thank you!